### ISSUE TITLE
Can not catch error from reader

### ISSUE THREAD

**Issue by agirault** [InsightSoftwareConsortium/itk-js#75]
I get the following crash error when reading a file that is not supported by `readImageFile` (for example a txt or md file):
```
Uncaught DOMException: Failed to execute 'postMessage' on 'DedicatedWorkerGlobalScope':
Error: Could not find IO for: ${fileName} could not be cloned.
```

See below two ways to try to gracefully handle that error:

```js
import itkReadImageFile from 'itk/readImageFile';

itkReadImageFile(file).then(
  (image) => {
    console.log(`${file} successfuly read`);
  },
  (error) => {
    console.log(`Could not read ${file}: ${error}`);
  }
);
```
or
```js
import itkReadImageFile from 'itk/readImageFile';

itkReadImageFile(file)
  .then((image) => {
    console.log(`${file} successfuly read`);
  })
  .catch((error) => {
    console.log(`Could not read ${file}: ${error}`);
  });
```

References:
* [Using promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises)
* [Two ways to catch issues](https://stackoverflow.com/questions/33278280/promise-then-vs-then-catch)

**Comment by thewtex** [comment]
Todos:

- Create a test that reproduces this issue
- Look into replacing `promise-worker-transferrable` with [webworker-promise)(https://github.com/kwolfy/webworker-promise), which also has the nice ability to use an inlined web worker

**Comment by thewtex** [comment]
Addressed by #78
